fabric-datafixer-v1/src/main/java/net/fabricmc/fabric/impl/datafixer/FabricDataFixerImpl.java

```diff
@@ -0,0 +1,217 @@
+		}
+
+		return currentTag;
+	}
+
+	@Override
+	public boolean isLocked() {
+		return locked;
+	}
+
+	/**
+	 * @deprecated for implementation only.
+	 */
+	public void lock() {
+		if (!locked) {
+			LOGGER.info("Locked DataFixer registration");
+		}
+
+		this.locked = true;
+	}
+
+	private static Schema createSchema() {
+		LOGGER.debug("Creating Fabric Schema Type");
+		List<DataFixerEntrypoint> entrypoints = FabricLoader.getInstance().getEntrypoints("fabric:datafixer", DataFixerEntrypoint.class);
+		LOGGER.debug("Found " + entrypoints.size() + " entrypoints.");
+		Schema schema = new Schema(0, VanillaDataFixers.VANILLA_DATAFIXER.apply(-1, null)) {
+			@Override
+			public void registerTypes(Schema schema, Map<String, Supplier<TypeTemplate>> entityTypes, Map<String, Supplier<TypeTemplate>> blockEntityTypes) {
+				super.registerTypes(schema, entityTypes, blockEntityTypes);
+
+				for (DataFixerEntrypoint entrypoint : entrypoints) {
+					LOGGER.debug("Registering types from " + entrypoint.getClass().getCanonicalName());
+					entrypoint.registerTypes(schema, entityTypes, blockEntityTypes);
+				}
+			}
+
+			@Override
+			public Map<String, Supplier<TypeTemplate>> registerBlockEntities(Schema schema) {
+				Map<String, Supplier<TypeTemplate>> map = super.registerBlockEntities(schema);
+
+				for (DataFixerEntrypoint entrypoint : entrypoints) {
+					LOGGER.debug("Registering BlockEntities from " + entrypoint.getClass().getCanonicalName());
+					entrypoint.registerBlockEntities(schema, map);
+				}
+
+				return map;
+			}
+
+			@Override
+			public Map<String, Supplier<TypeTemplate>> registerEntities(Schema schema) {
+				Map<String, Supplier<TypeTemplate>> map = super.registerEntities(schema);
+
+				for (DataFixerEntrypoint entrypoint : entrypoints) {
+					LOGGER.debug("Registering Entities from " + entrypoint.getClass().getCanonicalName());
+					entrypoint.registerEntities(schema, map);
+				}
+
+				return map;
+			}
+		};
+		return schema;
+	}
+
+	/**
+	 * An Entry which stores DataFixers for use by implementation.
+	 */
+	final class DataFixerEntry {
+		private final DataFixer dataFixer;
+		private final int runtimeDataVersion;
+
+		DataFixerEntry(DataFixer fix, int runtimeDataVersion) {
+			this.dataFixer = fix;
+			this.runtimeDataVersion = runtimeDataVersion;
+		}
+	}
+
+	/**
+	 * Represents Minecraft's Built in DataFixer.
+	 */
+	public static final class VanillaDataFixers {
+		private static final int LATEST_VANILLA_SCHEMA_VERSION = DataFixUtils.makeKey(SharedConstants.getGameVersion().getWorldVersion());
+
+		static {
+			LOGGER.info("Started with Vanilla-DataFixer version: " + LATEST_VANILLA_SCHEMA_VERSION);
+		}
+
+		public static final BiFunction<Integer, Schema, Schema> VANILLA_DATAFIXER = (version, parent) -> Schemas.getFixer().getSchema(LATEST_VANILLA_SCHEMA_VERSION);
```

Comment:
should put the lambda body in a static method declaration than let the compiler generate a method (see how java does `Objects#equals`)